socs/agents/srs_cg635/agent.py

```diff
@@ -148,12 +148,18 @@ def acq(self, session, params):
                     }
 
                     try:
-                        freq, stdc, runs, timb = self.clock.get_all_status()
-                        data['data']['Frequency'] = freq
-                        data['data']['Standard_CMOS_Output'] = stdc
-                        data['data']['Running_State'] = runs
-                        data['data']['Timebase'] = timb
-
+                        try:
+                            freq, stdc, runs, timb = self.clock.get_all_status().split(';')
+                            data['data']['Frequency'] = float(freq)
+                            data['data']['Standard_CMOS_Output'] = int(stdc)
+                            data['data']['Running_State'] = int(runs)
+                            data['data']['Timebase'] = int(timb)
+                        except ValueError:
+                            self.clock.clear()
+                            data['data']['Frequency'] = self.clock.get_freq()
+                            data['data']['Standard_CMOS_Output'] = self.clock.get_stdc()
+                            data['data']['Running_State'] = self.clock.get_runs()
+                            data['data']['Timebase'] = self.clock.get_timebase()
```

Comment:
This works, but I think would be better suited to go down a level, into the `SRSCG635Interface.get_all_status()` method. That way the driver is robust to this error, rather than just the agent.